src/quacc/atoms/skzcam.py

```diff
@@ -13,12 +13,489 @@
+        The coordinates block for the adsorbate-slab complex, the adsorbate, and the slab.
+    """
+
+    # First generate the preamble block
+    preamble_block = generate_orca_input_preamble(
+        embedded_adsorbed_cluster,
+        quantum_cluster_indices,
+        element_info,
+        pal_nprocs_block,
+        method_block,
+        scf_block,
+    )
+
+    # Generate the coords block
+    ad_slab_coords, ad_coords, slab_coords = generate_coords_block(
+        embedded_adsorbed_cluster,
+        quantum_cluster_indices,
+        ecp_region_indices,
+        ecp_info,
+        include_cp,
+        multiplicity,
+    )
+
+    # Combine the blocks
+    ad_slab_block = preamble_block + ad_slab_coords
+    ad_block = preamble_block + ad_coords
+    slab_block = preamble_block + slab_coords
+
+    return ad_slab_block, ad_block, slab_block
+
+
+def create_atom_coord_string(
+    atom: Atom,
+    ghost_atom: bool = False,
+    atom_ecp_info: str | None = None,
+    pc_charge: float | None = None,
+) -> str:
+    """
+    Creates a string containing the Atom symbol and coordinates in the ORCA input file format, with additional information for atoms in the ECP region as well as ghost atoms.
+
+    Parameters
+    ----------
+    atom
+        The ASE Atom (not Atoms) object containing the atomic coordinates.
+    atom_ecp_info
+        If not None, then assume this is an atom in the ECP region and add the ECP info.
+    ghost_atom
+        If True, then the atom is a ghost atom.
+    pc_charge
+        The point charge value for the ECP region atom.
+
+    Returns
+    -------
+    str
+        The atom symbol and coordinates in the ORCA input file format.
+    """
+
+    # If ecp_info is not None and ghost_atom is True, raise an error
+    if atom_ecp_info is not None and ghost_atom:
+        raise ValueError("ECP info cannot be provided for ghost atoms.")
+
+    # Check that pc_charge is a float if atom_ecp_info is not None
+    if atom_ecp_info is not None and pc_charge is None:
+        raise ValueError("Point charge value must be given for atoms with ECP info.")
+
+    if ghost_atom:
+        atom_coord_str = f"{(atom.symbol + ':').ljust(3)} {' '*16} {atom.position[0]:-16.11f} {atom.position[1]:-16.11f} {atom.position[2]:-16.11f}\n"
+    elif atom_ecp_info is not None:
+        atom_coord_str = f"{(atom.symbol + '>').ljust(3)} {pc_charge:-16.11f} {atom.position[0]:-16.11f} {atom.position[1]:-16.11f} {atom.position[2]:-16.11f}\n{atom_ecp_info}"
+    else:
+        atom_coord_str = f"{atom.symbol.ljust(3)} {' '*16} {atom.position[0]:-16.11f} {atom.position[1]:-16.11f} {atom.position[2]:-16.11f}\n"
+
+    return atom_coord_str
+
+
+def generate_coords_block(
+    embedded_adsorbed_cluster: Atoms,
+    quantum_cluster_indices: list[int],
+    ecp_region_indices: list[int],
+    ecp_info: dict[str, str] | None = None,
+    include_cp: bool = True,
+    multiplicity: dict[str, int] | None = None,
+) -> tuple[str, str, str]:
```

Comment:
Pretty much the same comments here about clarifying the type hints with `ElementStr`, `multiplicity` to `multiplicities`, the return type hint docstring, and clarifying in the docstring that things like `quantum_cluster_indices` are coming from the skzcam function.

Reply:
Thanks for these comments. Would you mind clarifying the issue with the return type hint docstring?

Reply:
Sorry my bad, Github puts some of your middle comments away and I didn't know I have to expand it to see all of them. I now know what to do!